File: beeCrowd/AlfabetoAlienigena/AlfabetoAlienigena.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>

void apresentacao(){
    printf("\nAlfabeto Alienigena\n\n");
}

void lerKeys(char* keys, int* Ktam){
    printf("Digite as keys: ");
    scanf("%s", keys);  

    *Ktam = 0; 
    while (keys[*Ktam] != '\0'){  
        (*Ktam)++;
    }
}

void lerFrase(char* frases, int* Ftam){
    char temp[100];
    printf("Digite a frase: ");
    fgets(temp, sizeof(temp), stdin);

    int j = 0;
    for(int i = 0; temp[i] != '\0'; i++){
        if (temp[i] != ' ' && temp[i] != '\n'){  
            frases[j] = temp[i]; 
            j++;
        }
    }
    frases[j] = '\0';  
    *Ftam = j;  
}

void keysNaFrase(char* frases, char* keys, int Ftam, int Ktam){
    
    int iguais = 0;
    for(int i = 0; i < Ktam; i++){
        for(int j = 0; j < Ftam; j++){
            if(keys[i] == frases[j]){  
                iguais++;
            }
        }
    }

    printf("%d\n", iguais);
}

int main() {
    apresentacao();

    char keys[100];
    char frases[100];
    int Ktam, Ftam;

    lerKeys(keys, &Ktam);

    while (getchar() != '\n'); 
    
    lerFrase(frases, &Ftam);

    keysNaFrase(frases, keys, Ftam, Ktam);

    return 0;
}

File: beeCrowd/AlfabetoAlienigena/AlfabetoAlienigena.java
import java.util.*;

public class AlfabetoAlienigena{

    public static void apresentacao(){
        System.out.println("\nAlfabeto Alienigena");
    }

    public static String lerKey(){
        
        Scanner scanner = new Scanner(System.in);
        System.out.printf("\nDigite uma key: ");
        String key = scanner.nextLine();

        return key;
    }

    public static String lerFrase(){

        Scanner scanner = new Scanner(System.in);
        System.out.printf("\nDigite uma frase: ");
        String frase = scanner.nextLine();

        scanner.close();

        return frase;
    }

    public static int comparar(String key, String frase){
        
        int iguais = 0;

        for(int i = 0; i < key.length(); i++){
            for(int j = 0; j < frase.length(); j++){
                if(key.charAt(i) == frase.charAt(j)){
                    iguais++;
                }
            }
        }

        return iguais;

    }
     public static void main(String[] args) {
        
        apresentacao();

        String key = lerKey();
        String frase = lerFrase();

        int iguais = comparar(key, frase);
    
    
        System.out.println("\nNúmero de caracteres iguais: " + iguais);
    }

}